plugin/trino-hive/src/main/java/io/trino/plugin/hive/ViewReaderUtil.java

```diff
@@ -69,7 +87,35 @@ public static ViewReader createViewReader(SemiTransactionalHiveMetastore metasto
             return new LegacyHiveViewReader();
         }
 
-        return new HiveViewReader(new CoralSemiTransactionalHiveMSCAdapter(metastore, new HiveIdentity(session)), typemanager);
+        return new HiveViewReader(
+                new CoralSemiTransactionalHiveMSCAdapter(metastore, new HiveIdentity(session), coralTableRedirection(session, tableRedirection, metadataProvider)),
+                typeManager);
+    }
+
+    private static CoralTableRedirection coralTableRedirection(
+            ConnectorSession session,
+            BiFunction<ConnectorSession, SchemaTableName, Optional<CatalogSchemaTableName>> tableRedirection,
+            MetadataProvider metadataProvider)
+    {
+        return schemaTableName -> tableRedirection.apply(session, schemaTableName).map(target -> {
```

Comment:
```suggestion
        return schemaTableName -> tableRedirection.apply(session, schemaTableName).map(target -> {
ConnectorRelationMetadata tableMetadata = metadataProvider.getRelationMetadata(session, target)
                    .orElseThrow(() -> new TableNotFoundException(
                            target.getSchemaTableName(),
                            format("%s is redirected to %s, but that relation cannot be found", schemaTableName, target)));
            List<Column> columns = tableMetadata.getColumns().stream()
                    .map(columnMetadata -> new Column(columnMetadata.getName(), toHiveType(columnMetadata.getType()), Optional.ofNullable(columnMetadata.getComment())))
                    .collect(toImmutableList());
            Table table = Table.builder()
                    .setDatabaseName(schemaTableName.getSchemaName())
                    .setTableName(schemaTableName.getTableName())
                    .setTableType(EXTERNAL_TABLE.name())
                    .setDataColumns(columns)
                    // Required by 'Table', but not used by view translation.
                    .withStorage(storage -> storage.setStorageFormat(fromHiveStorageFormat(TEXTFILE)))
                    .setOwner(Optional.empty())
                    .build();
            return toMetastoreApiTable(table);
        });        
        
```

```
        return schemaTableName -> tableRedirection.apply(session, schemaTableName)
                .map(catalogSchemaTableName -> getTable(session, metadataProvider, schemaTableName, catalogSchemaTableName))
                .map(ThriftMetastoreUtil::toMetastoreApiTable);
    }

    private static Table getTable(ConnectorSession session, MetadataProvider metadataProvider, SchemaTableName schemaTableName, CatalogSchemaTableName target)
    {
        ConnectorRelationMetadata tableMetadata = metadataProvider.getRelationMetadata(session, target)
                .orElseThrow(() -> new TableNotFoundException(
                        target.getSchemaTableName(),
                        format("%s is redirected to %s, but that relation cannot be found", schemaTableName, target)));
        List<Column> columns = tableMetadata.getColumns().stream()
                .map(columnMetadata -> new Column(columnMetadata.getName(), toHiveType(columnMetadata.getType()), Optional.ofNullable(columnMetadata.getComment())))
                .collect(toImmutableList());
        return Table.builder()
                .setDatabaseName(schemaTableName.getSchemaName())
                .setTableName(schemaTableName.getTableName())
                .setTableType(EXTERNAL_TABLE.name())
                .setDataColumns(columns)
                // Required by 'Table', but not used by view translation.
                .withStorage(storage -> storage.setStorageFormat(fromHiveStorageFormat(TEXTFILE)))
                .setOwner(Optional.empty())
                .build();
    }
```

Reply:
code cosmetics suggestion

Reply:
the diff in https://github.com/trinodb/trino/pull/10212#discussion_r765716311 seems to show exact same code as currently committed. what did you mean here?

Reply:
```
return schemaTableName -> tableRedirection.apply(session, schemaTableName)
                .map(catalogSchemaTableName -> getTable(session, metadataProvider, schemaTableName, catalogSchemaTableName))
                .map(ThriftMetastoreUtil::toMetastoreApiTable);
```

I meant to  slightly rewrite the code in a matter that the logic previously present in the anonymous class, to be moved to a separate method.

Reply:
i think it's fine as is